### PR TITLE
Effective Area Update

### PR DESCRIPTION
**Pull request by ehsteve**
Added thermal blankets, shutters, and detector efficiency to the effective area calculation.

### CODE REVIEW

File: idl/response/foxsi_get_detector_efficiency.pro [hunk 1]
```diff
@@ -0,0 +1,36 @@
+;+
+; NAME : foxsi_get_detector_efficiency
+;
+; PURPOSE : Returns the detector efficiency as a percent.
```
Comment (aringlis) [inline idl/response/foxsi_get_detector_efficiency.pro]: It says percent here, but presumably it should be fraction.

File: idl/response/foxsi_get_shutter_transmission.pro [hunk 2]
```diff
@@ -0,0 +1,45 @@
+;+
+; NAME : get_foxsi_get_shutter_transmission
+;
+; PURPOSE : Returns the shutter transmission efficiency.
+;
+; SYNTAX : trans = get_foxsi_get_shutter_transmission()
```
Comment (aringlis) [inline idl/response/foxsi_get_shutter_transmission.pro]: Ok this is a very minor quibble, but you can't actually call this with no arguments (shutter state is required)

File: idl/foxsi-smex-setup-script.pro [hunk 3]
```diff
@@ -13,6 +14,11 @@ foxsi_name = 'FOXSI-SMEX'
 foxsi_number_of_modules = 3
 foxsi_launch_date = '2020/06/01'
 
+foxsi_shutters_thickness_mm = [0, 1.0, 1.5]
+foxsi_detector_thickness_mm = 0.500
+; the blankets are assumed to be made of mylar
+foxsi_blanket_thickness_mm = 0.5
+
 ; the shells that are included in a FOXSI optics module
 foxsi_shell_ids = indgen(40)+1
```
Comment (aringlis) [inline idl/foxsi-smex-setup-script.pro:24]: shouldn't there be an END at the end of this script? I can't run it without adding that.

File: idl/response/foxsi_get_xray_transmission.pro [hunk 4]
```diff
@@ -0,0 +1,70 @@
+; NAME : foxsi_get_xray_transmission
+;
+; PURPOSE : Returns the x-ray transmission efficiency through a material
+;
+; SYNTAX : trans = foxsi_get_xray_transmission()
+;
+; INPUTS :
+;           thickness_mm - the thickness in mm
+;           material - material name as a string
```
Comment (aringlis) [inline idl/response/foxsi_get_xray_transmission.pro]: Maybe list allowed values of material somewhere.

**Comment by aringlis** [comment]
I pulled this branch and tested the code - it looks good. I had a few minor comments, but otherwise :+1:

File: idl/foxsi-smex-setup-script.pro (hunk 3 above)
Comment (ehsteve) [inline idl/foxsi-smex-setup-script.pro:24]: You need to run it with the @ symbol. See the readme

Comment (aringlis) [inline idl/foxsi-smex-setup-script.pro:24]: Meriem showed me that this morning - in all my years of using IDL I never knew that :-O

**Pull request merged**